# Comparing The 2010s SF Giants Championship Teams

I will be exploring and visualizing the regular season performance of the San Francisco Giants for their **World Series Championships** in 2010, 2012 and 2014.

In [1]:
from pybaseball import schedule_and_record
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
%matplotlib inline

In [2]:
from plotly import __version__
from plotly.offline import download_plotlyjs, init_notebook_mode, plot, iplot

import cufflinks as cf
init_notebook_mode(connected=True)
cf.go_offline()

In [3]:
sf2010 = schedule_and_record(2010, 'SF')
sf2012 = schedule_and_record(2012, 'SF')
sf2014 = schedule_and_record(2014, 'SF')

C:\Users\josh\anaconda3\lib\site-packages\pybaseball\team_results.py:70: FutureWarning:

In a future version of pandas all arguments of DataFrame.drop except for the argument 'labels' will be keyword-only.

C:\Users\josh\anaconda3\lib\site-packages\pybaseball\team_results.py:83: FutureWarning:

In a future version of pandas all arguments of DataFrame.drop except for the argument 'labels' will be keyword-only.

C:\Users\josh\anaconda3\lib\site-packages\pybaseball\team_results.py:70: FutureWarning:

In a future version of pandas all arguments of DataFrame.drop except for the argument 'labels' will be keyword-only.

C:\Users\josh\anaconda3\lib\site-packages\pybaseball\team_results.py:83: FutureWarning:

In a future version of pandas all arguments of DataFrame.drop except for the argument 'labels' will be keyword-only.

C:\Users\josh\anaconda3\lib\site-packages\pybaseball\team_results.py:70: FutureWarning:

In a future version of pandas all arguments of DataFrame.drop except for the argume

## Summarizing each of the Seasons

At first glance, the run differential appears to be the greatest for the 2010 team. No team had a particularly amazing winning streak, as the 2012 and 2014 teams' longest was 6 games and the 2010 team's was 5 games.

In [4]:
sf2010.describe()

R          RA         Inn        Rank    Attendance  \
count  162.000000  162.000000  162.000000  162.000000    162.000000   
mean     4.302469    3.598765    9.246914    2.172840  34766.166667   
std      3.293301    2.692201    0.804195    0.874627   9035.760383   
min      0.000000    0.000000    9.000000    1.000000  11633.000000   
25%      2.000000    2.000000    9.000000    2.000000  30346.750000   
50%      3.500000    3.000000    9.000000    2.000000  36247.500000   
75%      6.000000    5.000000    9.000000    3.000000  41834.000000   
max     16.000000   13.000000   15.000000    4.000000  53381.000000   

           Streak  
count  162.000000  
mean     0.302469  
std      2.170066  
min     -7.000000  
25%     -1.000000  
50%      1.000000  
75%      2.000000  
max      5.000000

In [5]:
sf2012.describe()

R          RA         Inn       Rank    Attendance      Streak
count  162.000000  162.000000  162.000000  162.00000    161.000000  162.000000
mean     4.432099    4.006173    9.154321    1.67284  37320.173913    0.506173
std      2.904006    2.956196    0.616250    0.81774   7217.498871    2.022381
min      0.000000    0.000000    9.000000    1.00000  12835.000000   -5.000000
25%      2.000000    2.000000    9.000000    1.00000  32810.000000   -1.000000
50%      4.000000    3.000000    9.000000    2.00000  41157.000000    1.000000
75%      6.000000    6.000000    9.000000    2.00000  41791.000000    2.000000
max     16.000000   17.000000   14.000000    5.00000  56000.000000    6.000000

In [6]:
sf2014.describe()

R          RA         Inn        Rank    Attendance  \
count  162.000000  162.000000  162.000000  162.000000    162.000000   
mean     4.104938    3.790123    9.166667    1.456790  37262.580247   
std      3.010557    2.822782    0.689315    0.511954   7787.672236   
min      0.000000    0.000000    9.000000    1.000000  13675.000000   
25%      2.000000    2.000000    9.000000    1.000000  31329.500000   
50%      4.000000    3.000000    9.000000    1.000000  41169.000000   
75%      6.000000    5.000000    9.000000    2.000000  41724.750000   
max     15.000000   17.000000   14.000000    3.000000  53500.000000   

           Streak  
count  162.000000  
mean     0.246914  
std      2.411313  
min     -6.000000  
25%     -1.000000  
50%      1.000000  
75%      2.000000  
max      6.000000

## Plotting records over the course of the season

All three teams followed a fairly similar path over the course of the season and end at around 90 wins each time. The 2014 team started out the hottest but had a smaller increase in wins after around 80 games and ended up with the worst regular season record out of the 3 teams. The plot is interactive so you can see the exact number of wins the teams had throughout each season.

In [7]:
sf2010.head()
# Taking a closer look at the W/L column, there are four unique values: W, L, W-wo, L-wo. (wo stands for walk off)
# To count the wins throughout the season in the next section, I need to count both W and W-wo.

Date   Tm Home_Away  Opp   W/L     R   RA   Inn  W-L  Rank  \
1     Monday, Apr 5  SFG         @  HOU     W   5.0  2.0   9.0  1-0   1.0   
2    Tuesday, Apr 6  SFG         @  HOU     W   3.0  0.0   9.0  2-0   1.0   
3  Wednesday, Apr 7  SFG         @  HOU     W  10.0  4.0   9.0  3-0   1.0   
4     Friday, Apr 9  SFG      Home  ATL  W-wo   5.0  4.0  13.0  4-0   1.0   
5  Saturday, Apr 10  SFG      Home  ATL     L   2.0  7.0   9.0  4-1   1.0   

       GB       Win        Loss    Save  Time D/N  Attendance   cLI  Streak  \
1    Tied  Lincecum      Oswalt  Wilson  2:34   N     43836.0   .96       1   
2  up 1.0      Zito   Rodriguez  Wilson  2:38   N     24237.0  1.04       2   
3  up 1.0   Affeldt    Gervacio    None  3:01   D     21599.0  1.09       3   
4  up 1.0   Affeldt      Medlen    None  4:01   D     42940.0  1.12       4   
5  up 1.0      Lowe  Wellemeyer    None  3:06   N     42985.0  1.12      -1   

  Orig. Scheduled  
1            None  
2            None  
3            None  
4            None  
5            None

In [8]:
# New dataframe for win count
sfWinCount = pd.DataFrame()

In [9]:
# Taking the cumulative sum of wins to date at each point in the season
sfWinCount['2010 Wins'] = np.where((sf2010['W/L']=='W') | (sf2010['W/L'] == 'W-wo'), 1, 0).cumsum()
sfWinCount['2012 Wins'] = np.where((sf2012['W/L']=='W') | (sf2012['W/L'] == 'W-wo'), 1, 0).cumsum()
sfWinCount['2014 Wins'] = np.where((sf2014['W/L']=='W') | (sf2014['W/L'] == 'W-wo'), 1, 0).cumsum()

In [10]:
sfWinCount.iplot(title = 'SF Giants Record Throughout Season',xTitle='Games Played',yTitle='Number of Wins')

## Final Standings

Generally, a team with around 90 wins should win their divison. By checking the standings for each of the three seasons, I can see if each team won the divison or entered the playoffs via the Wild Card. The 2010 and 2012 teams, with records above 90 games, won the division each year. The 2014 team with 87 wins snuck into the playoffs with a Wild Card spot.

In [11]:
from pybaseball import standings

In [12]:
# I am taking an index because the 2010 standings item is a list of dataframes and the 5th index represents the NL West
standings(2010)[5]

Tm   W   L  W-L%    GB
1  San Francisco Giants  92  70  .568    --
2      San Diego Padres  90  72  .556   2.0
3      Colorado Rockies  83  79  .512   9.0
4   Los Angeles Dodgers  80  82  .494  12.0
5  Arizona Diamondbacks  65  97  .401  27.0

In [13]:
standings(2012)[5]

Tm   W   L  W-L%    GB
1  San Francisco Giants  94  68  .580    --
2   Los Angeles Dodgers  86  76  .531   8.0
3  Arizona Diamondbacks  81  81  .500  13.0
4      San Diego Padres  76  86  .469  18.0
5      Colorado Rockies  64  98  .395  30.0

In [14]:
standings(2014)[5]

Tm   W   L  W-L%    GB
1   Los Angeles Dodgers  94  68  .580    --
2  San Francisco Giants  88  74  .543   6.0
3      San Diego Padres  77  85  .475  17.0
4      Colorado Rockies  66  96  .407  28.0
5  Arizona Diamondbacks  64  98  .395  30.0

## Score Differential

Below we can see the run differential for each of the championship teams. Run differential is calculated by runs scored - runs allowed. The 2010 team outscored their opponents by the widest margin at 114 runs. The 2014 team, which finished with the least number of wins, also finished with the worst run differential at just 51 runs. Their rough end to the season is also evident here, as they had a run differential of 67 runs just 64 games into the season and finished with a worse run differential despite playing almost 100 more games by the end of the season. Still, they made the playoffs and were able to use their championship experience to win their third World Series in 5 years.

In [15]:
# New dataframe for score differential
sfDiff = pd.DataFrame()

In [16]:
sfDiff['2010 Differential'] = (sf2010['R'] - sf2010['RA']).cumsum()
sfDiff['2012 Differential'] = (sf2012['R'] - sf2012['RA']).cumsum()
sfDiff['2014 Differential'] = (sf2014['R'] - sf2014['RA']).cumsum()

In [17]:
sfDiff.iplot(title = 'SF Giants Run Differential Throughout Season',xTitle='Games Played',yTitle='Run Differential')

## Conclusion

Solely looking at regular season performance, the 2010 San Francisco Giants seem to be the best of the three World Series winning teams in the 2010s. Although the 2012 team had a better regular season record, they only had 2 more wins than the 2010 team and the 2010 team had a better run differential by 45 runs. However, a championship is a championship and all 3 of the teams here got it done.

## Additional Exploration

I also want to investigate the big name players on the Giants and how they compared to the MLB's best in each of their World Series winning seasons.

## The 2010 Season

In [18]:
from pybaseball import pitching_stats

In [19]:
pitching2010 = pitching_stats(2010)

In [20]:
# The 2010 Giants pitching rotation with key stats
pitching2010[['Name','Team','W','L','ERA','WHIP','IP','SO','K/BB','K/9','FIP','WAR']][pitching2010['Team'] == 'SFG']

Name Team   W   L   ERA  WHIP     IP   SO  K/BB   K/9   FIP  \
29      Tim Lincecum  SFG  16  10  3.43  1.27  212.1  231  3.04  9.79  3.15   
19         Matt Cain  SFG  13  11  3.14  1.08  223.1  177  2.90  7.13  3.65   
17  Jonathan Sanchez  SFG  13   9  3.07  1.23  193.1  205  2.14  9.54  4.00   
61        Barry Zito  SFG   9  14  4.15  1.34  199.1  150  1.79  6.77  4.25   

    WAR  
29  4.3  
19  4.0  
17  2.2  
61  1.8

In [21]:
# The 10 best pitchers with key stats
pitching2010[['Name','Team','W','L','ERA','WHIP','IP','SO','K/BB','K/9','FIP','WAR']].head(10)

Name   Team   W   L   ERA  WHIP     IP   SO   K/BB   K/9  \
20          Cliff Lee  - - -  12   9  3.18  1.00  212.1  185  10.28  7.84   
27   Justin Verlander    DET  18   9  3.37  1.16  224.1  219   3.08  8.79   
0     Felix Hernandez    SEA  13  12  2.27  1.06  249.2  232   3.31  8.36   
4        Roy Halladay    PHI  21  10  2.44  1.04  250.2  219   7.30  7.86   
10     Ubaldo Jimenez    COL  19   8  2.88  1.15  221.2  214   2.33  8.69   
1        Josh Johnson    FLA  11   6  2.30  1.11  183.2  186   3.88  9.11   
15       Jered Weaver    LAA  13  12  3.01  1.07  224.1  233   4.31  9.35   
36  Francisco Liriano    MIN  14  10  3.62  1.26  191.2  201   3.47  9.44   
21        CC Sabathia    NYY  21   7  3.18  1.19  237.2  197   2.66  7.46   
3     Adam Wainwright    STL  20  11  2.42  1.05  230.1  213   3.80  8.32   

     FIP  WAR  
20  2.58  7.3  
27  2.97  6.7  
0   3.04  6.7  
4   3.01  6.2  
10  3.10  6.0  
1   2.41  6.0  
15  3.06  5.9  
36  2.66  5.6  
21  3.54  5.5  
3   2.86  5.1

The best pitcher for the 2010 Giants, Tim Lincecum, posted a WAR of 4.3. The top 10 pitchers in the league had a WAR that varied from 5.1 to 7.3. Although this Giants team did not have a top tier starter, Tim Lincecum and Matt Cain posted very respectable WARs.

In [22]:
from pybaseball import batting_stats

In [23]:
batting2010 = batting_stats(2010)

In [24]:
# The 2010 Giants best hitters with key stats
batting2010[['Name','Team','AVG','OBP','SLG','OPS','RBI','HR','wOBA','OBP+','wRC','WAR']][batting2010['Team'] == 'SFG']

Name Team    AVG    OBP    SLG    OPS  RBI  HR   wOBA  OBP+  \
47    Andres Torres  SFG  0.268  0.343  0.479  0.823   63  16  0.358   103   
14      Aubrey Huff  SFG  0.290  0.385  0.506  0.891   86  26  0.387   116   
96       Juan Uribe  SFG  0.248  0.310  0.440  0.749   85  24  0.324    93   
112  Pablo Sandoval  SFG  0.268  0.323  0.409  0.732   63  13  0.316    97   

     wRC  WAR  
47    82  6.3  
14   112  5.7  
96    67  2.7  
112   68  1.1

In [25]:
# The 10 best hitters with key stats
batting2010[['Name','Team','AVG','OBP','SLG','OPS','RBI','HR','wOBA','OBP+','wRC','WAR']].head(10)

Name Team    AVG    OBP    SLG    OPS  RBI  HR   wOBA  OBP+  \
0    Josh Hamilton  TEX  0.359  0.411  0.633  1.044  100  32  0.445   125   
34   Carl Crawford  TBR  0.307  0.356  0.495  0.851   90  19  0.369   109   
26   Evan Longoria  TBR  0.294  0.372  0.507  0.879  104  22  0.376   113   
1       Joey Votto  CIN  0.324  0.424  0.600  1.024  113  37  0.438   128   
4    Albert Pujols  STL  0.312  0.414  0.596  1.011  118  42  0.419   125   
44    Brian McCann  ATL  0.269  0.375  0.453  0.828   77  21  0.362   113   
12  Ryan Zimmerman  WSN  0.307  0.388  0.510  0.899   85  25  0.390   117   
3    Jose Bautista  TOR  0.260  0.378  0.617  0.995  124  54  0.422   115   
11   Adrian Beltre  BOS  0.321  0.365  0.553  0.919  102  28  0.391   111   
10   Robinson Cano  NYY  0.319  0.381  0.534  0.914  109  29  0.391   116   

    wRC  WAR  
0   122  8.4  
34  101  7.7  
26  105  7.5  
1   135  6.9  
4   135  6.8  
44   83  6.7  
12  102  6.6  
3   134  6.5  
11  109  6.4  
10  119  6.4

Andres Torres and Aubrey Huff were the key players for the 2010 Giants by WAR, and Andres Torres was close to being one of the top 10 position players in baseball.

## The 2012 Season

In [26]:
pitching2012 = pitching_stats(2012)

In [27]:
# The 2012 Giants pitching rotation with key stats
pitching2012[['Name','Team','W','L','ERA','WHIP','IP','SO','K/BB','K/9','FIP','WAR']][pitching2012['Team'] == 'SFG']

Name Team   W   L   ERA  WHIP     IP   SO  K/BB   K/9   FIP  \
5           Matt Cain  SFG  16   5  2.79  1.04  219.1  193  3.78  7.92  3.40   
19  Madison Bumgarner  SFG  16  11  3.37  1.11  208.1  191  3.90  8.25  3.50   
18     Ryan Vogelsong  SFG  14   9  3.37  1.23  189.2  158  2.55  7.50  3.70   
61         Barry Zito  SFG  15   8  4.15  1.39  184.1  114  1.63  5.57  4.49   
84       Tim Lincecum  SFG  10  15  5.18  1.47  186.0  190  2.11  9.19  4.18   

    WAR  
5   3.2  
19  2.7  
18  1.9  
61  1.0  
84  0.9

In [28]:
# The 10 best pitchers with key stats
pitching2012[['Name','Team','W','L','ERA','WHIP','IP','SO','K/BB','K/9','FIP','WAR']].head(10)

Name   Team   W  L   ERA  WHIP     IP   SO  K/BB    K/9   FIP  \
2   Justin Verlander    DET  17  8  2.64  1.06  238.1  239  3.98   9.03  2.94   
12   Felix Hernandez    SEA  13  9  3.06  1.14  232.0  223  3.98   8.65  2.84   
0    Clayton Kershaw    LAD  14  9  2.53  1.02  227.2  229  3.63   9.05  2.89   
8       Gio Gonzalez    WSN  21  8  2.89  1.13  199.1  207  2.72   9.35  2.82   
14         Cliff Lee    PHI   6  9  3.16  1.11  211.0  207  7.39   8.83  3.13   
26      Zack Greinke  - - -  15  5  3.48  1.20  212.1  200  3.70   8.48  3.10   
3        R.A. Dickey    NYM  20  6  2.73  1.05  233.2  230  4.26   8.86  3.27   
10        Chris Sale    CHW  17  8  3.05  1.14  192.0  192  3.76   9.00  3.27   
51        Yu Darvish    TEX  16  9  3.90  1.28  191.1  221  2.48  10.40  3.29   
11       Cole Hamels    PHI  17  6  3.05  1.12  215.1  216  4.15   9.03  3.30   

    WAR  
2   6.9  
12  6.6  
0   6.1  
8   5.0  
14  5.0  
26  4.8  
3   4.7  
10  4.7  
51  4.7  
11  4.6

The best pitcher for the 2012 Giants, Matt Cain, posted a WAR of 3.2. The top 10 pitchers in the league had a WAR that varied from 4.6 to 6.9. Tim Lincecum is no longer a key starter like he was two years prior. Although Madison Bumgarner's stats are good, it is surprising to not see numbers similar to the top pitchers in the league because he was one of the best pitchers in the 2012 playoffs.

In [29]:
batting2012 = batting_stats(2012)

In [30]:
# The 2012 Giants best hitters with key stats
batting2012[['Name','Team','AVG','OBP','SLG','OPS','RBI','HR','wOBA','OBP+','wRC','WAR']][batting2012['Team'] == 'SFG']

Name Team    AVG    OBP    SLG    OPS  RBI  HR   wOBA  OBP+  wRC  \
3   Buster Posey  SFG  0.336  0.408  0.549  0.957  103  24  0.406   125  114   
73   Angel Pagan  SFG  0.288  0.338  0.440  0.778   56   8  0.334   103   85   

     WAR  
3   10.1  
73   4.6

In [31]:
# The 10 best hitters with key stats
batting2012[['Name','Team','AVG','OBP','SLG','OPS','RBI','HR','wOBA','OBP+','wRC','WAR']].head(10)

Name Team    AVG    OBP    SLG    OPS  RBI  HR   wOBA  OBP+  \
2         Mike Trout  LAA  0.326  0.399  0.564  0.963   83  30  0.409   125   
3       Buster Posey  SFG  0.336  0.408  0.549  0.957  103  24  0.406   125   
18     Yadier Molina  STL  0.315  0.373  0.501  0.874   76  22  0.375   114   
0     Miguel Cabrera  DET  0.330  0.393  0.606  0.999  139  44  0.417   123   
4   Andrew McCutchen  PIT  0.327  0.400  0.553  0.953   96  31  0.403   122   
7      Robinson Cano  NYY  0.313  0.379  0.550  0.929   94  33  0.394   118   
13     Chase Headley  SDP  0.286  0.376  0.498  0.875  115  31  0.378   115   
1         Ryan Braun  MIL  0.319  0.391  0.595  0.987  112  41  0.413   120   
15      David Wright  NYM  0.306  0.391  0.492  0.883   93  21  0.376   120   
8      Adrian Beltre  TEX  0.321  0.359  0.561  0.921  102  36  0.388   112   

    wRC   WAR  
2   121  10.1  
3   114  10.1  
18   92   7.7  
0   137   7.3  
4   124   7.3  
7   124   7.3  
13  115   7.2  
1   131   6.8  
15  109   6.6  
8   113   6.3

Buster Posey ascended into a superstar in the 2012 season, posting the best WAR in the league. He was an instrumental part of all their championship runs.

## The 2014 Season

In [32]:
pitching2014 = pitching_stats(2014)

In [33]:
# The 2014 Giants pitching rotation with key stats
pitching2014[['Name','Team','W','L','ERA','WHIP','IP','SO','K/BB','K/9','FIP','WAR']][pitching2014['Team'] == 'SFG']

Name Team   W   L   ERA  WHIP     IP   SO  K/BB   K/9   FIP  \
20  Madison Bumgarner  SFG  18  10  2.98  1.09  217.1  219  5.09  9.07  3.05   
47         Tim Hudson  SFG   9  13  3.57  1.23  189.1  120  3.53  5.70  3.54   
66     Ryan Vogelsong  SFG   8  13  4.00  1.28  184.2  151  2.60  7.36  3.85   

    WAR  
20  3.6  
47  1.7  
66  0.9

In [34]:
# The 10 best pitchers with key stats
pitching2014[['Name','Team','W','L','ERA','WHIP','IP','SO','K/BB','K/9','FIP','WAR']].head(10)

Name   Team   W   L   ERA  WHIP     IP   SO   K/BB    K/9  \
0     Clayton Kershaw    LAD  21   3  1.77  0.86  198.1  239   7.71  10.85   
6        Corey Kluber    CLE  18   9  2.44  1.09  235.2  269   5.27  10.27   
41        Phil Hughes    MIN  16  10  3.52  1.13  209.2  186  11.63   7.98   
25       Max Scherzer    DET  18   5  3.15  1.18  220.1  252   4.00  10.29   
1     Felix Hernandez    SEA  15   6  2.14  0.92  236.0  248   5.39   9.46   
28        David Price  - - -  15  12  3.26  1.08  248.1  271   7.13   9.82   
7          Jon Lester  - - -  16  11  2.46  1.10  219.2  220   4.58   9.01   
11  Jordan Zimmermann    WSN  14   5  2.66  1.07  199.2  182   6.28   8.20   
2          Chris Sale    CHW  12   4  2.17  0.97  174.0  208   5.33  10.76   
4     Adam Wainwright    STL  20   9  2.38  1.03  227.0  179   3.58   7.10   

     FIP  WAR  
0   1.81  7.9  
6   2.35  7.2  
41  2.65  6.3  
25  2.85  5.6  
1   2.56  5.6  
28  2.78  5.6  
7   2.80  5.4  
11  2.68  5.4  
2   2.57  5.0  
4   2.88  4.9

The best pitcher for the 2012 Giants, Madison Bumgarner, posted a WAR of 3.6, a slight improvement from the 2012 season. The top 10 pitchers in the league had a WAR that varied from 4.9 to 7.9. Bumgarner is the only pitcher with a WAR above 2 for the Giants. This weaker pitching staff makes sense as the 2014 Giants had the worst record and run differential of the 3 championship teams.

In [35]:
batting2014 = batting_stats(2014)

In [36]:
# The 2014 Giants best hitters with key stats
batting2014[['Name','Team','AVG','OBP','SLG','OPS','RBI','HR','wOBA','OBP+','wRC','WAR']][batting2014['Team'] == 'SFG']

Name Team    AVG    OBP    SLG    OPS  RBI  HR   wOBA  OBP+  \
15       Buster Posey  SFG  0.311  0.364  0.490  0.854   89  22  0.371   114   
48       Hunter Pence  SFG  0.277  0.332  0.445  0.777   74  20  0.341   104   
80     Pablo Sandoval  SFG  0.279  0.324  0.415  0.739   73  16  0.323   101   
105  Brandon Crawford  SFG  0.246  0.324  0.389  0.713   69  10  0.309   101   

     wRC  WAR  
15    93  7.6  
48    93  4.0  
80    75  3.2  
105   60  2.7

In [37]:
# The 10 best hitters with key stats
batting2014[['Name','Team','AVG','OBP','SLG','OPS','RBI','HR','wOBA','OBP+','wRC','WAR']].head(10)

Name Team    AVG    OBP    SLG    OPS  RBI  HR   wOBA  OBP+  \
4          Mike Trout  LAA  0.287  0.377  0.561  0.939  111  36  0.402   119   
19    Jonathan Lucroy  MIL  0.301  0.373  0.465  0.837   69  13  0.368   116   
15       Buster Posey  SFG  0.311  0.364  0.490  0.854   89  22  0.371   114   
0    Andrew McCutchen  PIT  0.314  0.410  0.542  0.952   83  25  0.412   128   
3   Giancarlo Stanton  MIA  0.288  0.395  0.555  0.950  105  37  0.403   123   
8    Michael Brantley  CLE  0.327  0.385  0.506  0.890   97  20  0.389   122   
26     Anthony Rendon  WSN  0.287  0.351  0.473  0.824   83  21  0.361   110   
5       Jose Bautista  TOR  0.286  0.403  0.524  0.928  103  35  0.402   127   
10      Adrian Beltre  TEX  0.324  0.388  0.492  0.879   77  19  0.380   123   
20       Carlos Gomez  MIL  0.284  0.356  0.477  0.833   73  23  0.368   111   

    wRC  WAR  
4   126  8.3  
19  100  8.2  
15   93  7.6  
0   120  7.4  
3   114  6.8  
8   113  6.5  
26  100  6.4  
5   120  6.2  
10   99  5.9  
20   98  5.7

Buster Posey is still one of the best in the big leagues with a WAR of 7.6, the third best in the league. Compared to 2012, this Giants team had more key hitters and their offense was likely the reason they were able to make the playoffs despite their pitching rotation.